dbms/src/IO/Compression/CompressedWriteBuffer.h

```diff
@@ -41,6 +48,18 @@ class CompressedWriteBuffer : public BufferWithOwnMemory<WriteBuffer>
         CompressionSettings compression_settings = CompressionSettings(),
         size_t buf_size = DBMS_DEFAULT_BUFFER_SIZE);
 
+    static std::unique_ptr<WriteBuffer> build(
```

Comment:
```suggestion
    static std::unique_ptr<CompressedWriteBuffer<true>> build(
```

return `std::unique_ptr<CompressedWriteBuffer<true>>` ?

Reply:
When `enable_legacy_checksum` is false, this function return `std::unique_ptr<CompressedWriteBuffer<false>>`, which can not be hold by `std::unique_ptr<CompressedWriteBuffer<true>>`

Reply:
ok